src/main/java/de/dhbw/karlsruhe/word/generation/WordGeneration.java

```diff
@@ -17,11 +18,76 @@ public class WordGeneration {
     private final float preferEndProductionProbabilityLimit = 0.9f;
     private int currentProductionCount;
     private int maxProductionCount;
+    private int maxReadCount;
+    private int currentReadCount;
+    private int maxActionCount;
+    private int currentActionCount;
 
     public WordGeneration(Grammar grammar){
         this.grammar = grammar;
     }
 
+    public String generateWord(ParserLimitation parserLimitation) throws WordLimitationsNotFulfillableException {
+        return generateWord(parserLimitation,1);
+    }
+
+    public String generateWord(ParserLimitation parserLimitation, int minWordLength) throws WordLimitationsNotFulfillableException{
+
+        this.maxReadCount = parserLimitation.maxReadCount;
+        this.maxActionCount = parserLimitation.maxActionCount;
+        List<GrammarProduction> startProductions = getPotentialStartProductions();
+
+        int countTries = 0;
+        String word;
+        do {
+            countTries++;
+            if (countTries>100){
+                throw new WordLimitationsNotFulfillableException();
+            }
+
+            currentReadCount = 0;
+            currentActionCount = 0;
+
+            GrammarProduction firstProduction = startProductions.get(rand.nextInt(startProductions.size()));
+            try {
+                word = traverseProductionsWithParserLimitations(firstProduction);
+                word = word.replace("ε", "");
+            } catch (ToManyProductionsException e){
+                word="";
+            }
+        } while (word.length() < minWordLength);
+        return word;
+    }
+
+    private String traverseProductionsWithParserLimitations(GrammarProduction production) throws ToManyProductionsException{
+
+        if (productionService.isEndProduction(production)) {
+            currentReadCount += production.rightSide().length();
+            if (currentReadCount > maxReadCount || currentActionCount > maxActionCount)
```

Comment:
_currentActionCount_ verändert sich in diesem if-Statement doch gar nicht. Muss dies trotzdem geprüft werden?

Reply:
Prüfung kann eventuell doch sinnvoll sein, falls jemand lustig ist und Zahlen < 0 reingibt für maxActionCount oder maxReadCount...